refresh applications-monitoring-centreon-mbi doc CTOR-604 (#4670)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-monitoring-centreon-mbi.md

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="DWH-db-content" label="DWH-db-content">

No metrics for this service.

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

No metrics for this service.

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

| Name | Unit |
|:-----|:-----|
| failedJob | count |

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Name                | Unit |
|:--------------------|:-----|
| time.offset.seconds | s    |

</TabItem>
<TabItem value="proc-cbis" label="proc-cbis">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |

</TabItem>
<TabItem value="proc-mysql" label="proc-mysql">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |

</TabItem>
</Tabs>

## Prerequisites

### Device Configuration

The configuration of SNMP on a Linux server is detailed in [the *Linux SNMP* Monitoring Connector's documentation page](operatingsystems-linux-snmp.md#net-snmp-server-configuration).

### Network flows

> The Centreon Poller must be able to reach UDP/161 (SNMP) and TCP/22 (SSH) ports of the central nodes.

Author: ```suggestion
The monitoring server must be able to reach the UDP/161 (SNMP) and TCP/5666 (NRPE) ports of the reporting server.
```

| WARNING      | Warning threshold of matching processes count                                                                                                  |               |           |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) | --memory      |           |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash

> /usr/lib/centreon/plugins/centreon_pacemaker_ssh.pl \

Author: ```suggestion
/usr/lib/centreon/plugins/centreon_linux_snmp.pl \
```

id: applications-monitoring-centreon-mbi
title: Centreon MBI
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

The following monitoring connectors will be installed when you install the **Centreon-MBI** connector through the

> **Configuration > Monitoring Connector Manager** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

* [Base Pack](./base-generic.md)
* [Linux SNMP](./operatingsystems-linux-snmp.md)

## Pack assets

### Templates

The Monitoring Connector **Centreon-MBI** brings a host template:

* **App-Monitoring-Centreon-MBI-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="App-Monitoring-Centreon-MBI-custom" label="App-Monitoring-MBI-custom">

| Alias          | Modèle de service                                 | Description                                                              |
|:---------------|:--------------------------------------------------|:-------------------------------------------------------------------------|
| DWH-db-content | App-Monitoring-Centreon-MBI-DWH-db-content-custom | Check if the datawarehouse of the reporting server is up to date         |
| DWH-partitions | App-Monitoring-Centreon-MBI-DWH-partitions-custom | Check if partitions of the reporting server datawarehouse are up to date |

> | failed-jobs    | App-Monitoring-Centreon-MBI-failed-jobs-custom    | Check if failed job are present                                          |

Reviewer: ```suggestion
| failed-jobs    | App-Monitoring-Centreon-MBI-failed-jobs-custom    | Check if failed jobs are present                                          |
```

| Ntp            | App-Monitoring-Centreon-MBI-Ntp-custom            | Check the synchronization with an NTP server via SNMP                    |

> | proc-cbis      | App-Monitoring-Centreon-MBI-Process-cbis-custom   | Check the presence and the memory consumption of cbis process via SNMP   |

Reviewer: ```suggestion
| proc-cbis      | App-Monitoring-Centreon-MBI-Process-cbis-custom   | Check the presence and the memory consumption of the cbis process via SNMP   |
```

> | proc-mysql     | App-Monitoring-Centreon-MBI-Process-mysql-custom  | Check the presence and the memory consumption of mysql process via SNMP  |

Reviewer: ```suggestion
| proc-mysql     | App-Monitoring-Centreon-MBI-Process-mysql-custom  | Check the presence and the memory consumption of the mysql process via SNMP  |
```

> The services listed above are created automatically when the **App-Monitoring-Centreon-MBI-custom** host template is used.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="DWH-db-content" label="DWH-db-content">

| Name | Unit |

> |:-----|:-----|

Reviewer: empty tables

Reviewer 2: Pas de métrique pour les deux premiers modes
Métrique failedJob (type numeric value pour le 3ème)

> |      |      |

Author: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

| Name | Unit |
|:-----|:-----|

> |      |      |

Author: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

| Name | Unit |
|:-----|:-----|

> |      |      |

Author: ```suggestion
| failedJob | count |
```

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Name                | Unit |
|:--------------------|:-----|
| time.offset.seconds | s    |

</TabItem>
<TabItem value="proc-cbis" label="proc-cbis">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |

</TabItem>
<TabItem value="proc-mysql" label="proc-mysql">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |

</TabItem>
</Tabs>

## Prerequisites

### Device Configuration

The configuration of SNMP on a Linux server is detailed in [the *Linux SNMP* Monitoring Connector's documentation page](operatingsystems-linux-snmp.md#net-snmp-server-configuration).

### Network flows

> The Centreon Poller must be able to reach UDP/161 (SNMP) and TCP/22 (SSH) ports of the central nodes.

Reviewer: ```suggestion
The Centreon Poller must be able to reach the UDP/161 (SNMP) and TCP/22 (SSH) ports of the central nodes.
```

Reviewer: of the reporting server? bad copy and paste

Author: Je te laisse proposer une suggestion parce que j'ai pas la réponse (en FR aussi)

Reviewer 2: La supervision du serveur MBI se fait via les protocoles SNMP et NRPE (dédié au contrôle du datawarehouse MBI).
Selon le cas elle est effectuée depuis le central (sans HA) ou depuis un poller (avec HA, le central ne faisant pas de supervision dans ce cas).
Dans ce cas la correction serait la suivante:
_The monitoring server must be able to reach the UDP/161 (SNMP) and TCP/5666 (NRPE) ports of the reporting server_.

## Installing the monitoring connector

### Pack

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-applications-monitoring-centreon-mbi centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-applications-monitoring-centreon-mbi centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-applications-monitoring-centreon-mbi centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-applications-monitoring-centreon-mbi centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Centreon-MBI** connector through

> the **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

```bash
apt install nagios-plugins-nrpe centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install nagios-plugins-nrpe centreon-pack-operatingsystems-linux-snmp
```

</TabItem>
</Tabs>

## Using the monitoring connector

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **App-Monitoring-Centreon-MBI-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro           | Description                                                                                                                                       | Default value       | Mandatory |
|:----------------|:--------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------|:---------:|

> | NRPECLIENT       | Name of the plugin to use to dialog with the NRPE daemon                                                                                          | check_centreon_nrpe |           |

Reviewer: ```suggestion
| NRPECLIENT       | Name of the plugin to use to talk with the NRPE daemon                                                                                          | check_centreon_nrpe |           |
```

> | NRPEEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [here](#available-options). | -u -m 8192          |           |

Reviewer: FR in EN

> | NRPEEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [here](#available-options). | -u -m 8192          |           |

Author: ```suggestion
| NRPEEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles). | -u -m 8192          |           |
```

| NRPEPORT         | TCP port the NRPE daemon is listening to                                                                                                          | 5666                |           |
| NRPETIMEOUT      | Command timeout                                                                                                                                   | 30                  |           |

> | SNMPEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [here](#available-options). |                     |           |

Reviewer: same

> | SNMPEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [here](#available-options). |                     |           |

Author: ```suggestion
| SNMPEXTRAOPTIONS | Any extra option you may want to add to every command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) |                     |           |
```

### Using a service template provided by the connector

1. If you have used a host template and checked **Create Services linked to the Template too**, the services linked to the template have been created automatically, using the corresponding service templates. Otherwise, [create manually the services you want](/docs/monitoring/basic-objects/services) and apply a service template to them.
2. Fill in the macros you want (e.g. to change the thresholds for the alerts). Some macros are mandatory (see the table below).

<Tabs groupId="sync">
<TabItem value="DWH-db-content" label="DWH-db-content">

| Macro        | Description                                                                                                                                    | Default value | Mandatory |
|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) |               |           |
```

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

| Macro        | Description                                                                                                                                    | Default value | Mandatory |
|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) |               |           |
```

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

| Macro        | Description                                                                                                                                    | Default value | Mandatory |
|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) |               |           |
```

|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| NTPADDR      | Set the ntp hostname (if not set, localtime is used)                                                                                           |               |           |
| NTPPORT      | Set the ntp port (default: 123)                                                                                                                |               |           |
| TIMEZONE     | Set the timezone of distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100'                                |               |           |
| WARNING      | Time offset warning threshold (in seconds)                                                                                                     | -1:1          |           |
| CRITICAL     | Time offset critical Threshold (in seconds)                                                                                                    | -2:2          |           |

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) |               |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) |               |           |
```

|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| PROCESSARGS  | Filter process arguments                                                                                                                       |               |           |
| PROCESSNAME  | Filter process name                                                                                                                            | ^java$        |           |
| PROCESSPATH  | Filter process path                                                                                                                            |               |           |
| CRITICAL     | Critical threshold of matching processes count                                                                                                 | 1:            |           |
| WARNING      | Warning threshold of matching processes count                                                                                                  |               |           |

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) | --memory      |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) | --memory      |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) | --memory      |           |
```

|:-------------|:-----------------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|
| PROCESSARGS  | Filter process arguments                                                                                                                       |               |           |
| PROCESSNAME  | Filter process name                                                                                                                            | ^mysqld| mariadb$  |             |
| PROCESSPATH  | Filter process path                                                                                                                            |               |           |
| CRITICAL     | Critical threshold of matching processes count                                                                                                 | 1:            |           |
| WARNING      | Warning threshold of matching processes count                                                                                                  |               |           |

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) | --memory      |           |

Reviewer: same

> | EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [here](#available-options) | --memory      |           |

Author: ```suggestion
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles) | --memory      |           |
```

| list-storages [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/liststorages.pm)]     | Not used in this Monitoring Connector                                                                       |
| load [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/loadaverage.pm)]               | Not used in this Monitoring Connector                                                                       |
| memory [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/memory.pm)]                  | Not used in this Monitoring Connector                                                                       |
| processcount [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/processcount.pm)]      | App-Monitoring-Centreon-MBI-Process-cbis-custom<br />App-Monitoring-Centreon-MBI-Process-mysql-custom<br /> |
| storage [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/storage.pm)]                | Not used in this Monitoring Connector                                                                       |

> | swap [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/swap.pm)]                      | ONot used in this Monitoring Connector                                                                      |

Author: ```suggestion
| swap [[code](https://github.com/centreon/centreon-plugins/blob/develop/src/snmp_standard/mode/swap.pm)]                      | Not used in this Monitoring Connector                                                                      |
```

#### Modes options

All available options for each service template are listed below:

<Tabs groupId="sync">
<TabItem value="DWH-db-content" label="DWH-db-content">

| Option   | Description   |
|:---------|:--------------|
|          |               |

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

| Option   | Description   |
|:---------|:--------------|
|          |               |

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

| Option   | Description   |
|:---------|:--------------|

> |          |               |

Author: ```suggestion
No options for this mode.

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

No options for this mode.

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

No options for this mode.
```

| proc-mysql     | App-Monitoring-Centreon-MBI-Process-mysql-custom  | Check the presence and the memory consumption of mysql process via SNMP  |

> The services listed above are created automatically when the **App-Monitoring-Centreon-MBI-custom** host template is used.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="DWH-db-content" label="DWH-db-content">

| Name | Unit |
|:-----|:-----|
|      |      |

</TabItem>
<TabItem value="DWH-partitions" label="DWH-partitions">

| Name | Unit |
|:-----|:-----|
|      |      |

</TabItem>
<TabItem value="failed-jobs" label="failed-jobs">

| Name | Unit |
|:-----|:-----|
|      |      |

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Name                | Unit |
|:--------------------|:-----|
| time.offset.seconds | s    |

</TabItem>
<TabItem value="proc-cbis" label="proc-cbis">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |
| mem_total | B    |
| mem_avg   | B    |

> | cpu_total | %    |

Author: ```suggestion
```

</TabItem>
<TabItem value="proc-mysql" label="proc-mysql">

| Name      | Unit |
|:----------|:-----|
| nbproc    | N/A  |
| mem_total | B    |
| mem_avg   | B    |

> | cpu_total | %    |

Author: ```suggestion
```
